sites/public/pages/listings.tsx

```diff
@@ -61,13 +61,17 @@ const ListingsPage = () => {
     if (page != currentPage || filters != filterState) {
       setCurrentPage(page)
       setFilterState(filters)
-      // TODO(abbiefarr): update the url with filter data (#240)
+      const query = { page: page }
+      for (const filterKey in filters) {
+        const filterValue = filters[filterKey]
+        if (filterValue) {
+          query[filterKey] = filterValue
+        }
+      }
       void router.push(
         {
           pathname: "/listings",
-          query: {
-            page: page,
-          },
+          query: query,
         },
         undefined,
         { shallow: true }
```

Comment:
This function is both updating the URL and updating the internal state, which I think is making this more difficult than it needs to be.

What if we split this into one that updates the URL given a `filterState` and another that updates the filterState given a query string?

Then, in the submit handler for the form, just updating the queryString. And then in the hook that listens for the querystring, we just update the filterstate, form state, and make a backend request.